### ISSUE TITLE
Wrong synID

### ISSUE THREAD

**Issue by sassanh** [pangloss/vim-javascript#945]
Look at this line:
```
_.map(_.filter(this.state.data.chapters, foo => true), bar => {});
```
when cursor is on foo syntax is: ['jsParen', 'jsArrowFuncArgs', 'jsFuncArgs']
when cursor is on bar syntax is: ['jsParen', 'jsArrowFuncArgs']
I guess the second is correct, and first is getting wrong highlight due to wrong syntax id.

**Comment by amadeus** [comment]
Oh yeah, something super messed up there, I'll take a look, thanks for the reproducible code snippet!

**Comment by amadeus** [comment]
Alright, I have a branch in place with a fix, but I want to sit on it for a bit: #946

**Comment by amadeus** [comment]
I've merged a fix for this into master
